server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -217,14 +216,9 @@ public Translog(
             logger.debug("open uncommitted translog checkpoint {}", checkpoint);
 
             final long minGenerationToRecoverFrom;
-            if (checkpoint.minTranslogGeneration < 0) {
-                final Version indexVersionCreated = indexSettings().getIndexVersionCreated();
-                assert indexVersionCreated.before(Version.V_6_0_0_beta1) :
-                    "no minTranslogGeneration in checkpoint, but index was created with version [" + indexVersionCreated + "]";
-                minGenerationToRecoverFrom = deletionPolicy.getMinTranslogGenerationForRecovery();
-            } else {
-                minGenerationToRecoverFrom = checkpoint.minTranslogGeneration;
-            }
+            assert checkpoint.minTranslogGeneration > 0 : "minTranslogGeneration < 0 should not happen if index was created with version"
```

Comment:
This doesn't seem to match the previous if condition (positive vs non-negative).. Also, why can't we just remove this assertion? It seemed to only exist to ensure a negative minTranslogGeneration occured with indices created before 6.0.

Reply:
I changed the assertion message. I read it like we now expect checkpoint.minTranslogGeneration to be non-negative. Not sure if this is worth checking but it also shouldn't hurt asserting this and might make the code more readable. If in doubt we can ping the distributed folks on this.